code_annotations/find_annotations.py

```diff
@@ -0,0 +1,254 @@
+
+    if source_path:
+        config['source_path'] = source_path
+
+    if report_path:
+        config['report_path'] = report_path
+
+    # This is a runtime option, shouldn't be in the config file
+    config['verbose'] = verbose
+
+    if verbose > 1:
+        print("Configuration:")
+        print(config)
+
+    return config
+
+
+def search(mgr, config):
+    """
+    Walks the source tree, sending known file types to the Stevedore
+    extensions
+
+    :param mgr: Stevedore NamedExtensionManager
+    :param config: Configuration dict
+    :return: Dict containing all of the search results
+    """
+
+    # Index the results by extension name
+    file_extensions_map = {}
+    known_extensions = set()
+    for extension_name in config['extensions']:
+        file_extensions_map[extension_name] = config['extensions'][extension_name]
+        known_extensions.update(config['extensions'][extension_name])
+
+    all_results = {}
+
+    for root, _, files in os.walk(config['source_path']):
+        for filename in files:
+            filename_extension = os.path.splitext(filename)[1][1:]
+
+            if filename_extension not in known_extensions:
+                if config['verbose'] > 2:
+                    print("{} is not a known extension, skipping ({}).".format(filename_extension, filename))
+                continue
+
+            full_name = os.path.join(root, filename)
+
+            if config['verbose'] > 2:
+                print(full_name)
+
+            # TODO: This should probably be a generator so we don't have to store all results in memory
+            with open(full_name, 'r') as file_handle:
+                # Call search_extension on all loaded extensions
+                results = mgr.map(search_extension, file_handle, file_extensions_map, filename_extension)
+
+                # Format and add the results to our running full set
+                format_file_results(config['source_path'], all_results, results)
+
+    return all_results
+
+
+def report(all_results, config):
+    """
+    Genrates the YAML report of all search results
+
+    :param all_results: Dict of found annotations, indexed by filename
+    :param config: Configuration dict
+    :return: Filename of generated report
+    """
+    if config['verbose'] > 1:
+        print(yaml.dump(all_results, default_flow_style=False))
+
+    now = datetime.datetime.now()
+    report_filename = os.path.join(config['report_path'], '{}.yaml'.format(now.strftime('%Y-%d-%m-%H-%M-%S')))
+
+    print("Generating report to {}".format(report_filename))
+
+    try:
+        os.makedirs(config['report_path'])
+    except OSError as e:
+        if e.errno != errno.EEXIST:
+            raise
+
+    with open(report_filename, 'w+') as report_file:
+        yaml.dump(all_results, report_file, default_flow_style=False)
+
+    return report_filename
+
+
+@click.command('static_find_annotations')
```

Comment:
With the `cli()` click group in cli.py, I envisioned implementing all the scripts as subcommands under that top level command.  How does `code_annotations static_find` sound, for instance?  Two advantages to this approach:

1. code_annotations --help gives an overview of all the commands
2. installing 1 package from pip adds 1 new name to the PATH.

Reply:
OTOH, some of that code scaffolding is still unmerged, so i'm actually okay with pushing this for now.

Reply:
Ok, does remind me that this should probably be using `click.echo` instead of prints to facilitate testing later too. I'll make that change and try to clean up the verbosity checks in the process.